packages/ui/src/ui/store/selectors/components/proxies/proxies.js

```diff
@@ -51,39 +35,54 @@ const getProxies = (state) => state.components.proxies.proxies.proxies;
 const getHostFilter = (state) => state.components.proxies.proxies.hostFilter;
 const getStateFilter = (state) => state.components.proxies.proxies.stateFilter;
 const getRoleFilter = (state) => state.components.proxies.proxies.roleFilter;
+const getBannedFilter = (state) => state.components.proxies.proxies.bannedFilter;
 const getSortState = (state) => state.tables[COMPONENTS_PROXIES_TABLE_ID];
 
-const getFilteredByHost = createSelector([getProxies, getHostFilter], (proxies, hostFilter) =>
-    _.filter(proxies, (proxy) => isCorrectHost(proxy, hostFilter)),
-);
-
-const getFilteredByState = createSelector(
-    [getFilteredByHost, getStateFilter],
-    (proxies, stateFilter) => _.filter(proxies, (proxy) => isCorrectState(proxy, stateFilter)),
-);
-
-const getFilteredByRole = createSelector(
-    [getFilteredByHost, getRoleFilter],
-    (proxies, roleFilter) => _.filter(proxies, (proxy) => isCorrectRole(proxy, roleFilter)),
+const getFiltersObject = createSelector(
+    [getHostFilter, getStateFilter, getRoleFilter, getBannedFilter],
+    (hostFilter, stateFilter, roleFilter, bannedFilter) => {
+        return {hostFilter, stateFilter, roleFilter, bannedFilter};
+    },
 );
 
 const getFilteredProxies = createSelector(
-    [getFilteredByHost, getFilteredByState, getFilteredByRole],
-    (filteredByHost, filteredByState, filteredByRole) =>
-        _.intersection(filteredByHost, filteredByState, filteredByRole),
+    [getProxies, getFiltersObject],
+    (proxies, filtersObject) => {
+        return filterProxies(proxies, filtersObject);
+    },
 );
 
+function filterProxies(proxies, {hostFilter, stateFilter, roleFilter, bannedFilter}) {
+    const bannedFilterAsBool = bannedFilter === 'true';
+    const predicates = _.compact([
+        hostFilter ? ({host}) => host?.toLowerCase().startsWith(hostFilter) : undefined,
+        stateFilter && stateFilter !== 'all' ? ({state}) => state === stateFilter : undefined,
+        roleFilter && roleFilter !== 'all' ? ({role}) => role === roleFilter : undefined,
+        bannedFilter !== 'all' ? ({banned}) => banned === bannedFilterAsBool : undefined,
+    ]);
+    return predicates.length ? _.filter(proxies, concatByAnd(...predicates)) : proxies;
+}
+
 const getAllRoles = createSelector([getProxies], (proxy) => aggregateItems(proxy, 'role'));
 
-const getVisibleRoles = createSelector([getFilteredByState], (proxy) =>
-    aggregateItems(proxy, 'role'),
+const getVisibleRoles = createSelector(
+    [getProxies, getFiltersObject],
+    function (
+        proxies,
+        {
+            // eslint-disable-next-line no-unused-vars
```

Comment:
`lodash/omit` is good when you don't know what fields should be removed. Better way is to rewrite the file with typescript and use native `_NAMING` of unused variables but it will involve too many other changes that I am not ready to do.